### PR TITLE
[VL] Add support for spark_partition_id function

### PR DESCRIPTION
**Pull request by gaoyangxiaozhu**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#ISSUE-ID)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecApiImpl.scala [hunk 1]
```diff
@@ -609,7 +609,9 @@ class SparkPlanExecApiImpl extends SparkPlanExecApi {
     Seq(
       Sig[HLLAdapter](ExpressionNames.APPROX_DISTINCT),
       Sig[UDFExpression](ExpressionNames.UDF_PLACEHOLDER),
-      Sig[NaNvl](ExpressionNames.NANVL))
+      Sig[NaNvl](ExpressionNames.NANVL),
+      Sig[SparkPartitionID](ExpressionNames.SPARK_PARTITION_ID)
```
Comment (PHILO-HE) [inline backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecApiImpl.scala:613]: See https://github.com/apache/incubator-gluten/pull/4954#discussion_r1525907862

Comment (gaoyangxiaozhu) [reply backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/SparkPlanExecApiImpl.scala:613]: will wait 4954 merged before update

**Comment by PHILO-HE** [comment]
Hi @gaoyangxiaozhu, I just found `montonically_increasing_id` falls back in this test: `SELECT montonically_increasing_id() from tbl;`, but no fallback for `SELECT a, montonically_increasing_id() from tbl`. Maybe, we need remove the fallback caused by `FallbackEmptySchemalRelation`? `spark_partition_id` should have the similar issue I think.

```
== Fallback Summary ==
(1) Scan parquet : at least one of its children has empty output
(2) Project: at least one of its children has empty output
(9) CollectLimit: Gluten does not touch it or does not support it

== Physical Plan ==
CollectLimit (9)
+- VeloxColumnarToRowExec (8)
   +- ^ ProjectExecTransformer (6)
      +- ^ InputIteratorTransformer (5)
         +- ^ InputAdapter (4)
            +- ^ RowToVeloxColumnar (3)
               +- ^ Project (2)
                  +- ^ Scan parquet  (1)

```

File: backends-velox/src/test/scala/io/glutenproject/execution/VeloxFunctionsValidateSuite.scala [hunk 2]
```diff
@@ -451,6 +451,13 @@ class VeloxFunctionsValidateSuite extends VeloxWholeStageTransformerSuite {
     }
   }
 
+  test("Test spark_partition_id function") {
+    runQueryAndCompare("""SELECT spark_partition_id(), l_orderkey
+                         | from lineitem limit 1""".stripMargin) {
```
Comment (PHILO-HE) [inline backends-velox/src/test/scala/io/glutenproject/execution/VeloxFunctionsValidateSuite.scala:456]: Let's use a larger value for `limit`, e.g., `limit 100`. We can also enlarge it for the above `monotonically_increasing_id` test.

Comment (gaoyangxiaozhu) [reply backends-velox/src/test/scala/io/glutenproject/execution/VeloxFunctionsValidateSuite.scala:456]: got! i will fix ahd upate for both functions in this PR later, thank you @PHILO-HE

Comment (gaoyangxiaozhu) [reply backends-velox/src/test/scala/io/glutenproject/execution/VeloxFunctionsValidateSuite.scala]: done

**Review by PHILO-HE — APPROVED** [review]
Thanks for your work!

**Pull request merged**